pkg/relayer/cmd/cmd.go

```diff
@@ -0,0 +1,250 @@
+	eventsquery "github.com/pokt-network/poktroll/pkg/client/events_query"
+	"github.com/pokt-network/poktroll/pkg/client/supplier"
+	"github.com/pokt-network/poktroll/pkg/client/tx"
+	"github.com/pokt-network/poktroll/pkg/relayer"
+	"github.com/pokt-network/poktroll/pkg/relayer/miner"
+	"github.com/pokt-network/poktroll/pkg/relayer/proxy"
+)
+
+var (
+	signingKeyName string
+	smtStorePath   string
+	sequencerNode  string
+	pocketNode     string
+)
+
+func RelayerCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		// TODO_DISCUSS: do we want to rename this to `relay-miner`?
+		Use:   "relayer",
+		Short: "Run a relayer",
+		Long:  `Run a relayer`,
+		RunE:  runRelayer,
+	}
+
+	cmd.Flags().String(cosmosflags.FlagKeyringBackend, "", "Select keyring's backend (os|file|kwallet|pass|test)")
+
+	// TECHDEBT: integrate these cosmosflags with the client context (i.e. cosmosflags, config, viper, etc.)
+	// This is simpler to do with server-side configs (see rootCmd#PersistentPreRunE).
+	// Will require more effort than currently justifiable.
+	cmd.Flags().StringVar(&signingKeyName, "signing-key", "", "Name of the key to sign transactions")
+	cmd.Flags().StringVar(&smtStorePath, "smt-store", "smt", "Path to the SMT KV store")
+	// Communication cosmosflags
+	// TODO_DISCUSS: We're using `explicitly omitting default` so the relayer crashes if these aren't specified. Figure out
+	// what the defaults should be post alpha.
+	cmd.Flags().StringVar(&sequencerNode, "sequencer-node", "explicitly omitting default", "<host>:<port> to sequencer/validator node to submit txs")
+	cmd.Flags().StringVar(&pocketNode, "pocket-node", "explicitly omitting default", "<host>:<port> to full/light pocket node for reading data and listening for on-chain events")
+	cmd.Flags().String(cosmosflags.FlagNode, "explicitly omitting default", "registering the default cosmos node flag; needed to initialize the cosmostx and query contexts correctly")
+
+	return cmd
+}
+
+func runRelayer(cmd *cobra.Command, _ []string) error {
+	ctx, cancelCtx := context.WithCancel(cmd.Context())
+	// Ensure context cancellation.
+	defer cancelCtx()
+
+	deps, err := setupRelayerDependencies(ctx, cmd)
+	if err != nil {
+		return err
+	}
+
+	var relayMiner relayer.RelayMiner
+	if err := depinject.Inject(
+		deps,
+		&relayMiner,
+	); err != nil {
+		return err
+	}
+
+	// Handle interrupts in a goroutine.
+	go func() {
+		sigCh := make(chan os.Signal, 1)
+		signal.Notify(sigCh, os.Interrupt)
+
+		// Block until we receive an interrupt or kill signal (OS-agnostic)
+		<-sigCh
+
+		// Signal goroutines to stop
+		cancelCtx()
+	}()
+
+	// Start the relay miner
+	log.Println("INFO: Starting relay miner...")
+	relayMiner.Start(ctx)
+
+	log.Println("INFO: Relay miner stopped; exiting")
+	return nil
+}
+
+func setupRelayerDependencies(
+	ctx context.Context,
+	cmd *cobra.Command,
+) (deps depinject.Config, err error) {
+	// Set --node flag to the --sequencer-node for the client context
```

Comment:
```suggestion
	// TODO_TECHDEBT: Revisit this decision and see if it can be removed
	// Set --node flag to the --sequencer-node for the client context
```